### ISSUE TITLE
[BUG] Upgrade fails on a k8s cluster that was previously scaled up

### ISSUE THREAD

**Issue by sk4zuzu** [epiphany-platform/epiphany#1473]
**Describe the bug**
Upgrade of the k8s Epiphany cluster fails if the number of worker nodes was previously increased. The "upgraded" inventory seems to be incorrectly modified (number of worker nodes is invalid).

**To Reproduce**
1. Deploy an older cluster (for example v0.5) with just 1 worker node.
2. Scale up the cluster increasing number of worker nodes to 3 (re-run `epicli apply`).
3. Switch to the develop branch.
4. Run the full upgrade of the cluster `epicli upgrade`.

**Expected behavior**
Cluster upgrades without problems. Ansible inventory is correct.

**Config files**
N/A

**OS (please complete the following information):**
 - All supported.

**Cloud Environment (please complete the following information):**
 - "Any" (currently is reproducible there, it's still to be determined for others)

**Additional context**
N/A

**Comment by mkyc** [comment]
@przemyslavic can you confirm that this bug exists?

**Comment by przemyslavic** [comment]
I will test it today for the following providers: `any`, `aws`, `azure`.

**Comment by przemyslavic** [comment]
I can't reproduce this issue.
Steps taken:
1. Deployed 0.5.4 cluster with 1 master and 1 node (using `epicli apply`)
2. Created 2 additional virtual machines
3. Modified yaml configuration file to have 3 nodes
4. Executed `epicli apply` again
5. Executed `epicli upgrade` (from the latest code from the develop branch)

The inventory file looks correct (contains entries for all nodes).
Providers that have been tested: `azure`, `aws`, `any`.
I tested it twice.
@sk4zuzu maybe in the meantime you took some extra steps that could cause this problem?

**Comment by sk4zuzu** [comment]
@przemyslavic that's actually good news. I tried this couple of times more and I lost ability to reproduce it either. I was able to reproduce it repeatedly on a cluster that I was recreating from backup / snapshots. My conclusion would be, that during testing that backup got broken in some way. What worries me is that our code was capable of creating invalid inventory and we don't have any verification for it. Closing for now. \\:D/
